clients/src/main/java/org/apache/kafka/clients/ClusterConnectionStates.java

```diff
@@ -42,7 +42,7 @@ final class ClusterConnectionStates {
     static final int CONNECTION_SETUP_TIMEOUT_EXP_BASE = 2;
     static final double CONNECTION_SETUP_TIMEOUT_JITTER = 0.2;
     private final Map<String, NodeConnectionState> nodeState;
-    private final Logger log;
+    private static final Logger log;
```

Comment:
Noted that this static variable should NOT be initialized in constructor.

Reply:
Thanks for finding the bug. This is the case that I shouldn't change the logger to static.
I have re-evaluated and used a different approach. I have completed the code updates and written a new PR message that includes the methodology and criteria used. PTAL.